uinput.go

```diff
@@ -0,0 +1,177 @@
+package evdev
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"syscall"
+	"time"
+)
+
+const (
+	uinputMaxNameSize = 80
+	absSize           = 64
+)
+
+// CreateDevice from scratch with the provided capabilities and name
+// If setup fails the device will be removed from the system,
+// once setup it can be removed by calling dev.Close
+func CreateDevice(name string, capabilities map[EvType][]EvCode) (*InputDevice, error) {
+	deviceFile, err := os.OpenFile("/dev/uinput", syscall.O_WRONLY|syscall.O_NONBLOCK, 0660)
+	if err != nil {
+		return nil, err
+	}
+
+	newDev := &InputDevice{
+		file: deviceFile,
+	}
+
+	for ev, codes := range capabilities {
+		if err := ioctlUISETEVBIT(newDev.file.Fd(), uintptr(ev)); err != nil {
+			DestroyDevice(newDev)
+			return nil, fmt.Errorf("failed to set ev bit: %d", ev)
+		}
+
+		if err := setEventCodes(newDev, ev, codes); err != nil {
+			DestroyDevice(newDev)
+			return nil, fmt.Errorf("failed to set ev code")
+		}
+	}
+
+	_, err = createUsbDevice(newDev.file, UinputUserDevice{
+		Name: toUinputName([]byte(name)),
+		ID: InputID{
+			BusType: 0x03,
+			Vendor:  0x4712,
+			Product: 0x0816,
+			Version: 1,
+		},
```

Comment:
I think you should pass this into the function as configuration block instead of hard-coding it.